File: Offer/1/Solution.java
public class Solution {
    public boolean Find(int target, int [][] array) {
       int m = array.length;
       if (m==0) return false;
        int n = array[0].length;
        if(n==0) return false;
        int r = 0, c = n - 1;
        while(r <m && c >=0){
            if(target == array[r][c]) return true;
            else if(target > array[r][c]) ++r;
            else if(target < array[r][c]) --c;
        }
        return false;
    }
}

File: Offer/1/Solution.go
package Solution

func find(target int, array [][]int) bool {

	m := len(array)
	if m == 0 {
		return false
	}
	n := len(array[0])
	if n == 0 {
		return false
	}
	r := 0
	c := n - 1
	for {
		if r > m-1 || c < 0 {
			break
		}
		if target == array[r][c] {
			return true
		} else if target > array[r][c] {
			r++
		} else if target < array[r][c] {
			c--
		}
	}
	return false

}
